internal/service/vpclattice/service_network_data_source.go

```diff
@@ -83,19 +92,27 @@ func dataSourceServiceNetworkRead(ctx context.Context, d *schema.ResourceData, m
 	d.Set("number_of_associated_vpcs", out.NumberOfAssociatedVPCs)
 	d.Set("service_network_identifier", out.Id)
 
+	return crossAccountSetTags(ctx, conn, diags, serviceNetworkARN, meta.(*conns.AWSClient).AccountID(ctx), "Service Network")
+}
+
+func crossAccountSetTags(ctx context.Context, conn *vpclattice.Client, diags diag.Diagnostics, resARN, accountID, resName string) diag.Diagnostics {
 	// https://docs.aws.amazon.com/vpc-lattice/latest/ug/sharing.html#sharing-perms
 	// Owners and consumers can list tags and can tag/untag resources in a service network that the account created.
 	// They can't list tags and tag/untag resources in a service network that aren't created by the account.
-	parsedARN, err := arn.Parse(serviceNetworkARN)
+	parsedARN, err := arn.Parse(resARN)
 	if err != nil {
 		return sdkdiag.AppendFromErr(diags, err)
 	}
 
-	if parsedARN.AccountID == meta.(*conns.AWSClient).AccountID(ctx) {
-		tags, err := listTags(ctx, conn, serviceNetworkARN)
+	if parsedARN.AccountID == accountID {
+		tags, err := listTags(ctx, conn, resARN)
+
+		if err != nil && errs.Contains(err, ErrCodeAccessDeniedException) {
```

Comment:
Can be simplified slightly.

```suggestion
		if errs.Contains(err, ErrCodeAccessDeniedException) {
```